jupyterhub/roles.py

```diff
@@ -75,35 +76,30 @@ def expand_self_scope(name):
 
 def _get_scope_hierarchy():
```

Comment:
Let's put a `@lru_cache()` on this one, since it doesn't change and is getting a bit more expensive.

Reply:
Thank you for the suggestion, which prompted the idea of just scrapping the whole hierarchy and it works!